pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -931,17 +933,38 @@ func (c *Reconciler) createCustomRun(ctx context.Context, runName string, params
 		Annotations:     getTaskrunAnnotations(pr),
 	}
 
+	var customRef *v1beta1.TaskRef
+	if rpt.PipelineTask.TaskRef != nil {
+		customRef = &v1beta1.TaskRef{}
+		customRef.ConvertFrom(ctx, *rpt.PipelineTask.TaskRef)
```

Comment:
The fields for v1beta1 CustomRun specs are needed to be converted back to V1beta1 from v1, so we would have to export those conversions for workspaces, params and customRef.

Reply:
If I understand correctly this is to make sure we don't have cyclic dependency as well right ?

Reply:
Could you please add comments in the code about this?